### PR TITLE
CL-1153 printer states in monitor

### PR DESCRIPTION
**Pull request by ianpaschal**
This brings the progress bar and printer state in line with the Cura 4.0 redesign

### CODE REVIEW

File: cura/PrinterOutput/PrintJobOutputModel.py [hunk 1]
```diff
@@ -118,17 +118,41 @@ def updateName(self, name: str):
...
+    def timeRemaining(self) -> int:
+        # Never get a negative time remaining
+        return max(self.timeTotal - self.timeElapsed, 0)
+
+    @pyqtProperty(float, notify = timeElapsedChanged)
+    def progress(self) -> float:
+        result = self.timeElapsed / self.timeTotal
+        if result > 1.0:
+            result = 1.0
+        return result
```
Comment (sedwards2009) [inline cura/PrinterOutput/PrintJobOutputModel.py]: It is not a  big deal, but it interesting to see `max()` used above but `min()` isn't being used here.

File: plugins/UM3NetworkPrinting/resources/qml/MonitorPrintJobPreview.qml [hunk 2]
```diff
@@ -47,11 +54,32 @@ Item
...
+                case "pausing":
+                    return "../svg/paused-icon.svg"
+                case "paused":
+                    return "../svg/paused-icon.svg"
+                case "resuming":
+                    return "../svg/paused-icon.svg"
+                default:
+                    return ""
+            }
+        }
```
Comment (sedwards2009) [inline plugins/UM3NetworkPrinting/resources/qml/MonitorPrintJobPreview.qml]: This function can return `undefined` when it should probably return something else like "".

File: plugins/UM3NetworkPrinting/resources/qml/MonitorPrintJobProgressBar.qml [hunk 3]
```diff
@@ -82,54 +29,85 @@ Item
...
+                case "pausing":
+                    return catalog.i18nc("@label:status", "Pausing...")
+                case "paused":
+                    return catalog.i18nc("@label:status", "Paused")
+                case "resuming":
+                    return catalog.i18nc("@label:status", "Resuming...")
+                case "queued":
+                    return catalog.i18nc("@label:status", "Action required")
+                default:
+                    return catalog.i18nc("@label:status", "Finishes ") + OutputDevice.getDateCompleted( printJob.timeRemaining ) + " at " + OutputDevice.getTimeCompleted( printJob.timeRemaining )
```
Comment (sedwards2009) [inline plugins/UM3NetworkPrinting/resources/qml/MonitorPrintJobProgressBar.qml]: This string building needs to be replaced with 1 translatable string/template.

File: cura/PrinterOutput/PrintJobOutputModel.py (hunk 1 above)
Comment (ianpaschal) [reply cura/PrinterOutput/PrintJobOutputModel.py]: Good point. Don't know why I didn't bother with that. Updated.

File: plugins/UM3NetworkPrinting/resources/qml/MonitorPrintJobPreview.qml (hunk 2 above)
Comment (ianpaschal) [reply plugins/UM3NetworkPrinting/resources/qml/MonitorPrintJobPreview.qml]: Ummmm can it? The switch is the first block and it has a default case at the end to ensure it always returns a string.

File: plugins/UM3NetworkPrinting/resources/qml/MonitorPrintJobProgressBar.qml (hunk 3 above)
Comment (sedwards2009) [reply plugins/UM3NetworkPrinting/resources/qml/MonitorPrintJobProgressBar.qml]: Maybe it is now time to move some of stuff to the Python side? The Python code is actually full of i18n strings too.

**Comment by sedwards2009** [comment]
The Windows CI server is borked and no one is around to fix the build slave thingy.

**Pull request merged**